### PR TITLE
Improve the existing DevPortal REST API to generate token using token-exchange grant type

### PR DESCRIPTION
**Pull request by vithu30**
This PR adds changes to improve the `POST /applications/{applicationId}/oauth-keys/{keyMappingId}/generate-token` Devportal REST API to generate token using token exchange grant type.

Request format:
```curl
curl --location --request POST 'https://localhost:9443/api/am/devportal/v2/applications/91e162e8-576b-402e-bff7-0c40c6a7d1e7/oauth-keys/a5eb6431-8c1b-4d97-976b-81aba133681e/generate-token' \
--header 'authorization: Bearer ${accessToken}' \
--data-raw '{
    "consumerSecret": "v2ZSpVrFSKo5sUlK9K1R6BZScdka",
    "validityPeriod": 3600,
    "revokeToken": null,
    "scopes": [],
    "grantType": "TOKEN_EXCHANGE",
    "additionalProperties": {
        "id_token_expiry_time": 3600,
        "application_access_token_expiry_time": 3600,
        "user_access_token_expiry_time": 3600,
        "refresh_token_expiry_time": 86400,
        "subject_token": "${externalIdPToken}"
    }
}'
```

Fixes https://github.com/wso2-enterprise/choreo/issues/5515

### CODE REVIEW

**Comment by Nirothipan** [comment]
do we not pass subject_token_type as a param here, even though now this is for jwt, won't it help when we extend?

**Review by pubudu538 — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIConsumer.java [hunk 1]
```diff
@@ -785,8 +785,8 @@ String getOpenAPIDefinitionForEnvironment(API api, String environmentName)
      * @throws APIManagementException
      */
     AccessTokenInfo renewAccessToken(String oldAccessToken, String clientId, String clientSecret, String validityTime,
-                                     String[] requestedScopes, String jsonInput,String keyManagerName) throws
-            APIManagementException;
+                                     String[] requestedScopes, String jsonInput, String keyManagerName,
+                                     String grantType) throws APIManagementException;
```
Comment (Nirothipan) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIConsumer.java:789]: shall we update the method comment

**Comment by vithu30** [comment]
> do we not pass subject_token_type as a param here, even though now this is for jwt, won't it help when we extend?

For our use case, it's not required to pass it from UI. If we extend it future we can pass it within the `additionalProperties`, hence it will not require any API changes.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/gen/java/org/wso2/carbon/apimgt/rest/api/store/v1/dto/ApplicationTokenGenerateRequestDTO.java [hunk 2]
```diff
@@ -26,6 +26,38 @@
...
+        }
+
+        @JsonCreator
+        public static GrantTypeEnum fromValue(String v) {
+            for (GrantTypeEnum b : GrantTypeEnum.values()) {
+                if (String.valueOf(b.value).equals(v)) {
+                    return b;
+                }
+            }
+return null;
```
Comment (Nirothipan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/gen/java/org/wso2/carbon/apimgt/rest/api/store/v1/dto/ApplicationTokenGenerateRequestDTO.java:57]: is this autogenerated, can't we format this

Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/gen/java/org/wso2/carbon/apimgt/rest/api/store/v1/dto/ApplicationTokenGenerateRequestDTO.java:57]: Yes, this is auto generated code

**Review by Nirothipan — APPROVED** [review]

**Pull request merged**